pulsar-client/src/main/java/org/apache/pulsar/client/api/CryptoKeyReader.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.client.api;
+
+public interface CryptoKeyReader {
+
+    /*
+     * Return the encryption key corresponding to the key name in the argument
+     * <p>
+     * This method should be implemented to return the key in byte array. This method will be
+     * called at the time of producer creation as well as consumer receiving messages.
+     * Hence, application should not make any blocking calls within the implementation. 
+     * While reading public key from the producer, the key name will have prefix "public-key." added to it.
+     * Similarly, to get private key from consumer, prefix "private-key." is used.
```

Comment:
just a thought : instead adding prefix business, should we have two apis
```
byte [] getPrivateKey(String keyName);
byte [] getPublicKey(String keyName);
```

Reply:
I like this proposal. @msb-at-yahoo Do you have any objection?